packages/mongodb-mql-model/src/docs/md/mql-query/mql-query.md

```diff
@@ -0,0 +1,145 @@
+
+* MongoDB Community 8.0, from a development environment, does not have an index on bookName for the target collection.
+* MongoDB Atlas 8.0, production environment, does have an index on bookName for the target collection.
+
+In the development environment, we copy the data from the production environment once every week. For this example,
+we will consider that the data sets are exactly the same on both clusters.
+
+| Cluster Environment           	 | Is Valid 	  | Is Supported 	 | Same Results 	 | Same Dialect 	 | Same Execution Plan 	 |
+|---------------------------------|-------------|----------------|----------------|----------------|-----------------------|
+| MongoDB Development         	   | ✅         	 | ✅        	     | ✅          	   | ✅        	     | 🔴                	   |
+| MongoDB Production 	            | ✅         	 | ✅           	  | ✅         	    | ✅        	     | 🔴                	   |
+
+**✅ They are equivalent.**
+
+And now, finally, let's assume the same environment, but with the query written in two dialects,
+`mongosh` and `Java Driver`:
+
+```java
+collection.find(eq("bookName", myBookName))
+```
+```js
+collection.find({"bookName": myBookName})
+```
+
+| Cluster Environment           	 | Is Valid 	  | Is Supported 	 | Same Results 	 | Same Dialect 	 | Same Execution Plan 	 |
+|---------------------------------|-------------|----------------|----------------|----------------|-----------------------|
+| MongoDB Development         	   | ✅         	 | ✅        	     | ✅          	   | 🔴        	    | 🔴                	   |
+| MongoDB Production 	            | ✅         	 | ✅           	  | ✅         	    | 🔴        	    | 🔴                	   |
+
+**✅ They are equivalent.**
+
+
+### Query Nodes
+
+A [MQL Node or a Node for short](/packages/mongodb-mql-model/src/main/kotlin/com/mongodb/jbplugin/mql/Node.kt) 
+represents a set of semantic properties of a MongoDB query or a subset of it. Nodes MUST NOT be specific to
+a single source dialect, but MAY contain semantics that are relevant for processing.
+
+Nodes MUST contain a single reference to the original source code, in any of the valid dialects. Multiple
+nodes MAY contain a reference to the same original source code. That reference is called the **source** 
+of the Node. For example, let's consider this query, written in the **Java Driver dialect** and how it is referenced by a Node.
+
+```java
+    collection.find(eq("_id", 123456)).first();
+//  ^                                        ^
+//  +----------------------------------------+
+//            Node(source)
+```
+
+A Node MAY contain parent nodes and children nodes, through specific **components**. A Node that
+doesn't contain any parent node, but contains children nodes is called the **root** node, and
+represents the whole query.
+
+Nodes MAY have additional components that contain metadata for that node. Components MAY have
+references to other Nodes and other components.
```

Comment:
We should clarify that components are an ordered list in our case, because I'd expect the typical behavior in a component pattern to be that of an unordered list